### PR TITLE
Use Snowflake IDs for nodes and edges

### PR DESCRIPTION
**Pull request by cmutel**
Needed for distributed object creation without conflicting id values.

### CODE REVIEW

**Comment by cmutel** [comment]
~Merging blocked by https://github.com/brightway-lca/matrix_utils/issues/22~

This is now fixed.

File: bw2data/backends/base.py
Comment (jsvgoncalves) [inline bw2data/backends/base.py]: This renaming will conflict with #174, as it uses these methods too.

File: bw2data/backends/schema.py
Comment (jsvgoncalves) [inline bw2data/backends/schema.py]: Also conflicting with #174; will need to be rebased after that one is merged. We can then make `SignaledDataset` inherit from this one and up goes the super.

File: bw2data/backends/schema.py
Comment (jsvgoncalves) [inline bw2data/backends/schema.py]: `--peewee;`

File: bw2data/backends/utils.py
Comment (jsvgoncalves) [inline bw2data/backends/utils.py]: Even though `add_snowflake_id` is optional, this can break #174, as it expects `dict_as_…` to have identical signatures. Will we ever need an `insert_many` via the `revisions` module?

File: bw2data/backends/proxies.py
Comment (jsvgoncalves) [inline bw2data/backends/proxies.py]:
```suggestion
                raise ValueError(f"`id` must be created automatically, but id={kwargs["key"]} was given.")
```

Slightly easier to debug with this message. Same for the exception below.

Comment (jsvgoncalves) [inline bw2data/backends/proxies.py]: Do we need the same validation for nodes? https://github.com/brightway-lca/brightway2-data/blob/b716b1420a1874cdaf139fae57494ec54beac994/bw2data/backends/base.py#L680

**Review by jsvgoncalves — COMMENTED** [review]
~Hold until #174 is merged~.

File: bw2data/backends/schema.py
Comment (cmutel) [reply bw2data/backends/schema.py]: > We can then make SignaledDataset 

exatamente

File: bw2data/backends/utils.py
Comment (cmutel) [reply bw2data/backends/utils.py]: > Will we ever need an insert_many via the revisions module?

Not in our current conception as each changeset is for one object at a time, and we apply them sequentially. But I think we will need to have custom behaviour for all `peewee` methods which change state on the database, and handle each one of them.

File: bw2data/backends/proxies.py
Comment (cmutel) [reply bw2data/backends/proxies.py]: Fixed in https://github.com/brightway-lca/brightway2-data/pull/202/commits/623f44135426a10d394fcd9834f7f35b428c660d

**Comment by jsvgoncalves** [comment]
After #174 is merged:
- [x] Fix schemas inheritance
- [x] Adjust `_dict_as_…`
- [x] Add snowflake implementation to revision ids

**Comment by cmutel** [comment]
>  Adjust `_dict_as_`…

This will go away as we will explicitly map functions instead of using string interpolation, see [more-event-types](https://github.com/brightway-lca/brightway2-data/commit/44b836f408a994595d15ac0a8f27a9b6f7710aa3) branch.

**Review by bbguimaraes — APPROVED** [review]

**Review by jsvgoncalves — APPROVED** [review]

**Pull request merged**
